pkg/vsphere/tags/rest_client.go

```diff
@@ -185,3 +192,17 @@ func (c *RestClient) Login(ctx context.Context) error {
 	Logger.Debugf("Login succeeded")
 	return nil
 }
+
+func (c *RestClient) newRequest(method, urlStr string, body io.Reader) (*http.Request, error) {
+	req, err := http.NewRequest(method, c.endpoint.String()+urlStr, body)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.user != nil {
+		password, _ := c.user.Password()
+		req.SetBasicAuth(c.user.Username(), password)
```

Comment:
At some point we should look into how we can re-use the sdk session cookie, rather than sending user+pass credentials again.  For example, the sdk cookie name is `vmware_soap_session`, the pbm and vsan API extensions accept the same cookie value for authn, but with a different cookie name: `vcSessionCookie`